File: qqbot-gateway/src/main/java/com/mieai/qqbot/gateway/GatewayCloseDecision.java
package com.mieai.qqbot.gateway;

import java.util.Objects;

/** Classified action for a WebSocket close. */
public record GatewayCloseDecision(
        int statusCode,
        GatewayCloseDisposition disposition,
        GatewayReconnectCause reconnectCause) {
    public GatewayCloseDecision {
        Objects.requireNonNull(disposition, "disposition must not be null");
        Objects.requireNonNull(reconnectCause, "reconnectCause must not be null");
    }
}

File: qqbot-gateway/src/main/kotlin/com/mieai/qqbot/gateway/GatewayCloseDecision.kt
package com.mieai.qqbot.gateway

/** Classified action for a WebSocket close. */
data class GatewayCloseDecision(
    val statusCode: Int,
    val disposition: GatewayCloseDisposition,
    val reconnectCause: GatewayReconnectCause,
)
